src/test/java/com/microsoft/util/ElementUtilTest.java

```diff
@@ -55,13 +55,23 @@ public void extractPackageElements() {
     public void extractSortedElements() {
         Element element = elements.getPackageElement("com.microsoft.samples.subpackage");
 
-        List<String> result = elementUtil.extractSortedElements(element)
+        List<String> allElements = element.getEnclosedElements()
+                .stream().map(String::valueOf).collect(Collectors.toList());
+
+        // Ensure items to exclude exist.
+        assertThat("Wrong enclosed elements number", allElements.size(), is(6));
+        assertTrue("Unexpected package private class", allElements.contains("com.microsoft.samples.subpackage.InternalException"));
+        assertTrue("Unexpected to-exclude class", allElements.contains("com.microsoft.samples.subpackage.SomeExcludedClass"));
+
+
+        List<String> extractedElements = elementUtil.extractSortedElements(element)
             .stream().map(String::valueOf).collect(Collectors.toList());
 
-        assertThat("Wrong result list size", result.size(), is(4));
-        assertThat("Unexpected first item", result.get(0), is("com.microsoft.samples.subpackage.CustomException"));
-        assertThat("Unexpected second item", result.get(1), is("com.microsoft.samples.subpackage.Display"));
-        assertThat("Unexpected third item", result.get(2), is("com.microsoft.samples.subpackage.Person"));
+        // Verify filtered and sorted result
+        assertThat("Wrong result list size", extractedElements.size(), is(4));
+        assertThat("Unexpected first item", extractedElements.get(0), is("com.microsoft.samples.subpackage.CustomException"));

```

Comment:
make it meaningful. For example, "expect subpackage class"

Reply:
Based on my understanding, using "Unexpected third item" here was to verified the orders, and print a helpful message if the test case failed.

I think the message can be refined a little.

Reply:
For exam, if the test fails here, you will get "unexpected first item", but I don't which is to expected, make it harder for developers other than you to debug.

Reply:
Here is an example message we will get from Maven

```
extractSortedElements(com.microsoft.util.ElementUtilTest)  Time elapsed: 0.149 sec  <<< FAILURE!
java.lang.AssertionError: Unexpected second item
Expected: is "com.microsoft.samples.subpackage.Display"
     but: was "com.microsoft.samples.subpackage.Person"
```

I am thinking change the message to "Unexpected second item in the result list after invoke method extractSortedElements()"

Reply:
Yes, that will be good enough.